pkg/reconciler/vertex/watcher.go

```diff
@@ -0,0 +1,25 @@
+package vertex
+
+import (
+	"fmt"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// getVertexStatus calculate the status by iterating over pods objects
+func getVertexStatus(vertex *dfv1.Vertex, pods *corev1.PodList) (string, string, bool) {
+	if len(pods.Items) == 0 {
+		return "No Pods found", "CooldownPeriod", true
+	} else if *vertex.Spec.Replicas != int32(len(pods.Items)) {
```

Comment:
Maybe we do not need to check this. It could be in transient status (e.g. rolling update).